### PR TITLE
prov/verbs: added iface-name filtering

### PR DESCRIPTION
**Pull request by hoopoepg**
on systems where multiple network interfaces configured
user may specify preferrable interface to be used by provider

this is additional fix for
https://github.com/ofiwg/libfabric/pull/2665

Signed-off-by: Oblomov, Sergey <sergey.oblomov@intel.com>

### CODE REVIEW

**Comment by JoZie** [comment]
I just did some testing of the PR on our machine and it worked as expected.
Thank you!

**Comment by hoopoepg** [comment]
Great!!!
Thank you for help

**Comment by shefty** [comment]
Is it possible to replace verbs specific interface filtering in favor of filtering based on the domain name?  Is a separate environment variable necessary?

**Comment by a-ilango** [comment]
@shefty domain name filtering should work (assuming different domains/devices come up as different interfaces), unless the user doesn't know which interface corresponds to which domain. For verbs this is not exactly obvious. I think one has to compare the interface HW address with GID. @hoopoepg might know if there were other reasons to have interface filtering.

@hoopoepg btw, does this PR contain changes from #2665 ? Can you please rebase over head of master?

**Comment by shefty** [comment]
Got it - thanks for the clarification.

File: prov/verbs/src/verbs_info.c [hunk 1]
```diff
@@ -890,18 +890,45 @@ static int fi_ibv_getifaddrs(struct dlist_entry *verbs_devs)
 	const char *ret_ptr;
 	int ret, num_verbs_ifs = 0;
 
+	char *iface = 0;
```
Comment (a-ilango) [inline prov/verbs/src/verbs_info.c]: This needs to be initialized to NULL instead of 0.

File: prov/verbs/src/verbs_info.c [hunk 2]
```diff
@@ -890,18 +890,45 @@ static int fi_ibv_getifaddrs(struct dlist_entry *verbs_devs)
...
+	char iface_len = 0;
```
Comment (a-ilango) [inline prov/verbs/src/verbs_info.c]: Shouldn't this be declared as size_t?

File: prov/verbs/src/verbs_info.c (hunk 1 above)
Comment (hoopoepg) [inline prov/verbs/src/verbs_info.c]: fixed

File: prov/verbs/src/verbs_info.c (hunk 2 above)
Comment (hoopoepg) [inline prov/verbs/src/verbs_info.c]: there is limit for iface name length = 16 bytes, that is why I used char... but you are right - size_t is better here.
fixed. thank you

**Pull request merged**